runtime/service/src/main/java/org/apache/polaris/service/exception/IcebergExceptionMapper.java

```diff
@@ -87,19 +87,27 @@ public class IcebergExceptionMapper implements ExceptionMapper<RuntimeException>
   private static final Set<String> ACCESS_DENIED_HINTS =
       Set.of("access denied", "not authorized", "forbidden");
 
-  public IcebergExceptionMapper() {}
+  private final Logger logger;
```

Comment:
Nit: instead of having a field, have a function that yields `LOGGER` in production code, but is overridden in tests. Should be easier for C2 to inline.